File: frontend/lib/vizard.ts
const BASE = "https://elb-api.vizard.ai/hvizard-server-front/open-api/v1";

function headers() {
  return {
    "Content-Type": "application/json",
    Authorization: `Bearer ${process.env.VIZARD_KEY ?? ""}`,
  };
}

export async function createProject(videoUrl: string, language = "ru") {
  if (!process.env.VIZARD_KEY) return { error: "VIZARD_KEY не задан" };
  try {
    const res = await fetch(`${BASE}/project/create`, {
      method: "POST",
      headers: headers(),
      body: JSON.stringify({
        videoUrl,
        language,
        autoClip: true,
        clipDuration: [30, 60, 90],
        virality: true,
      }),
    });
    return res.json();
  } catch (e: any) { return { error: e.message }; }
}

export async function pollProject(projectId: string) {
  if (!process.env.VIZARD_KEY) return { error: "VIZARD_KEY не задан" };
  try {
    const res = await fetch(`${BASE}/project/${projectId}`, { headers: headers() });
    return res.json();
  } catch (e: any) { return { error: e.message }; }
}

export async function getAiSocial(projectId: string) {
  if (!process.env.VIZARD_KEY) return { error: "VIZARD_KEY не задан" };
  try {
    const res = await fetch(`${BASE}/project/${projectId}/social`, { headers: headers() });
    return res.json();
  } catch (e: any) { return { error: e.message }; }
}

export async function publishVideoClip(projectId: string, caption: string, publishAt: string) {
  if (!process.env.VIZARD_KEY) return { error: "VIZARD_KEY не задан" };
  try {
    const res = await fetch(`${BASE}/project/${projectId}/publish`, {
      method: "POST",
      headers: headers(),
      body: JSON.stringify({ caption, publishAt: publishAt || undefined }),
    });
    return res.json();
  } catch (e: any) { return { error: e.message }; }
}

File: producer_center/api/vizard.py
"""Producer Center — Vizard API"""
import requests
from config import VIZARD_BASE, VIZARD_KEY

def _h():
    return {"Content-Type":"application/json","VIZARDAI_API_KEY": VIZARD_KEY}

def create_project(url: str, lang="ru", max_clips=10) -> dict:
    try:
        r = requests.post(f"{VIZARD_BASE}/project/create", headers=_h(), timeout=30,
            json={"videoUrl":url,"videoType":2 if "youtu" in url else 1,
                  "lang":lang,"preferLength":[2],"maxClipNumber":max_clips,
                  "ratioOfClip":4,"subtitleSwitch":1,"emojiSwitch":1,
                  "highlightSwitch":1,"headlineSwitch":1,"removeSilenceSwitch":1})
        return r.json()
    except Exception as e:
        return {"error": str(e)}

def poll_project(project_id) -> dict:
    try:
        r = requests.get(f"{VIZARD_BASE}/project/query/{project_id}",
                          headers=_h(), timeout=15)
        return r.json()
    except Exception as e:
        return {"error": str(e)}

def get_ai_social(video_id) -> dict:
    try:
        r = requests.post(f"{VIZARD_BASE}/project/ai-social", headers=_h(), timeout=20,
            json={"finalVideoId":video_id,"aiSocialPlatform":3,"tone":1,"voice":0})
        return r.json()
    except Exception as e:
        return {"error": str(e)}

def publish_video(video_id, caption: str, publish_time: int) -> dict:
    try:
        r = requests.post(f"{VIZARD_BASE}/project/publish-video", headers=_h(), timeout=15,
            json={"finalVideoId":video_id,"post":caption,"publishTime":publish_time})
        return r.json()
    except Exception as e:
        return {"error": str(e)}
